astropy/table/table.py

```diff
@@ -3869,26 +3870,17 @@ def _is_mixin_for_table(self, col):
 
     def _convert_col_for_table(self, col):
         if isinstance(col, Column) and getattr(col, 'unit', None) is not None:
-            # We need to turn the column into a quantity, or a subclass
-            # identified in the unit (such as u.mag()).
-            q_cls = getattr(col.unit, '_quantity_class', Quantity)
+            # We need to turn the column into a quantity; use subok=True to allow
+            # Quantity subclasses identified in the unit (such as u.mag()).
+            q_cls = Masked(Quantity) if isinstance(col, MaskedColumn) else Quantity
```

Comment:
Is there a plan to export a `MaskedQuantity` (`= Masked(Quantity)`) from `astropy.units`? I guess this is a high-level API question on whether users should use `MaskedQuantity` or `Masked(Quantity)`.

With the module level `__getattr__` you might be able to only lazy-load `MaskedQuantity` on demand, thus avoiding the need to import `Masked` in the `units.__init__.py`. Maybe...